### DISCUSSION TITLE
No dplr styles in lammps

### DISCUSSION THREAD

**Discussion by Constant206462** [deepmodeling/deepmd-kit#2314] [Q&A] [answered]
Hello. I just installed deepmd-kit v2.1.5 with cuda 11.6 using [this method](https://docs.deepmodeling.com/projects/deepmd/en/master/getting-started/install.html#install-off-line-packages). The problem is that `dplr` styles weren't set. When i run `lmp -h` command from created conda environment i don't see neither `pppm/dplr` among the kspace styles nor `dplr` among fixes. There was no errors or warnings during installation. Before that i used deepmd-kit v2.1.1 and there was no `dplr` styles as well (that's why i have installed newer version of deepmd-kit). 
So what should i do in order to use `dplr` in lammps calculations? I saw in Issues that it's a [bug](https://github.com/deepmodeling/deepmd-kit/issues/1567) and it seems like i have to build lammps from source with some [edited files](https://github.com/lammps/lammps/pull/3188/files), but can this be solved easily?

**Comment by njzjz** [comment, marked as answer]
`lmp -h` doesn't show the styles from plugins. You have to test manually.

**Reply by Constant206462** [reply]
Thank you for clarifying this part. Running lammps script with both deepmd versions 2.1.1 and 2.1.5 yields the same error tho.

> ERROR: Unrecognized kspace style 'pppm/dplr' (src/force.cpp:697)
Last command: kspace_style	pppm/dplr 1.0e-5

My lammps script if you need it.
[long_sim.txt](https://github.com/deepmodeling/deepmd-kit/files/10718537/long_sim.txt)

**Reply by njzjz** [reply]
Could you provide the log file?

**Reply by Constant206462** [reply]
Here
[from_2.1.1.txt](https://github.com/deepmodeling/deepmd-kit/files/10719862/slurm-11947.txt)
[from_2.1.5.txt](https://github.com/deepmodeling/deepmd-kit/files/10719861/slurm-11950.txt)

**Reply by Constant206462** [reply]
Oh wait a sec. The second one is not from 2.1.5

**Reply by Constant206462** [reply]
Well that's a shame on me.  I didn't change the path in my slutm file to 2.1.5 environment so it used the 2.1.1 one. It seems like the error I mentioned isn't occuring on 2.1.5. A bit wierd anyway that it's happening on 2.1.1. Sorry for disturbing.
